payjoin/src/core/receive/v2/mod.rs

```diff
@@ -381,9 +394,9 @@ impl Receiver<Initialized> {
         };
         match std::str::from_utf8(&body) {
             // V1 response bodies are utf8 plaintext
-            Ok(response) => Ok(Some(self.extract_proposal_from_v1(response)?)),
+            Ok(response) => Ok(Some(self.extract_proposal_from_v1(response).map(|p| (p, None))?)),
             // V2 response bodies are encrypted binary
-            Err(_) => Ok(Some(self.extract_proposal_from_v2(body)?)),
+            Err(_) => Ok(Some(self.extract_proposal_from_v2(body).map(|(p, e)| (p, Some(e)))?)),
```

Comment:
Nit: 
below you are using reply key and p. full variable name is more descriptive